Automattic/wp-calypso
# Stepper: Make framework tracking of submit-step the default. Updates remaining flows

Related to https://github.com/Automattic/wp-calypso/issues/91755

## Proposed Changes

* Makes framework tracking of submit-step the default. Any new flow will opt-in for tracking by the framework
* Migrates remaining flows that explicitly called `recordSubmitStep`
* Updates the following flows that hardcoded intent in the `recordSubmitStep` to pass it through the data-store: `readymade-template`, `link-in-bio-post-setup`, `newsletter-post-setup`, `update-design`, `update-options`
* We do not update/ensure that `intent` is provided and logged for all of these flows. Only for those that already recorded a value for it in their localised calls to `recordSubmitStep`

The PR touches several flows. We've tested individual migrations in previous PRs, so this follows along. 

## Why are these changes being made?


Part of addressing https://github.com/Automattic/wp-calypso/issues/91755

## Testing Instructions



* Code review. Ensure `recordSubmitStep` is (at least currently) only called from `useStepNavigationWithTracking`
* Confirm for any mentioned Stepper flow that `calypso_signup_actions_submit_step` is logged with correct values (in Netowrk tab, check for `.gif` request) when navigating from/to any step
* For the following flows, confirm that `intent` is logged correctly as the name of the flow: 
  * `/setup/readymade-template`
  * `/setup/link-in-bio-post-setup`
  * `/setup/newsletter-post-setup`
  * `/setup/update-design`
  * `/setup/update-options`

## Pre-merge Checklist
